cmd/create/cluster/cmd.go

```diff
@@ -862,6 +881,43 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	enableCustomerManagedKey := args.enableCustomerManagedKey
+	kmsKeyARN := args.kmsKeyARN
+
+	if interactive.Enabled() && kmsKeyARN == "" {
+		enableCustomerManagedKey, err = interactive.GetBool(interactive.Input{
+			Question: "Enable Customer Managed key",
+			Help:     cmd.Flags().Lookup("enable-customer-managed-key").Usage,
+			Default:  enableCustomerManagedKey,
+			Required: false,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid value for enable-customer-managed-key: %s", err)
+			os.Exit(1)
+		}
+	}
+	if kmsKeyARN != "" {
+		enableCustomerManagedKey = true
+	}
```

Comment:
Something still feels off about this. What if the user sets `--enable-customer-managed-key`, I don't think we want to show the prompt there... How about something like this:
```go
	if kmsKeyARN != "" {
		enableCustomerManagedKey = true
	}
	if interactive.Enabled() && !enableCustomerManagedKey {
		enableCustomerManagedKey, err = interactive.GetBool(interactive.Input{
			Question: "Enable Customer Managed key",
			Help:     cmd.Flags().Lookup("enable-customer-managed-key").Usage,
			Default:  enableCustomerManagedKey,
			Required: false,
		})
		if err != nil {
			reporter.Errorf("Expected a valid value for enable-customer-managed-key: %s", err)
			os.Exit(1)
		}
	}
```
This means that we would show the prompt only when on interactive mode and the user hasn't set either of the two new flags. If there is any of the new flags set, then we skip the first prompt and only ask for the key.

(sorry about all the nitpicking... `--enable-foo` flags often cause confusion during the interactive flow)

Reply:
@vkareh  this is what we are doing. show enable-customer-managed-key only when KMSkey arn is empty and in interactive mode.